tests/framework/results/test_context.py

```diff
@@ -154,6 +154,33 @@ def test_double_add_observation(
     assert len(ctx._observations) == 1
 
 
+@pytest.mark.parametrize(
+    "default, additional, excluded, match",
+    [
+        (["age", "sex"], ["age"], [], "age"),
+        (["age", "sex"], [], ["eye_color"], "eye_color"),
+        (["age", "sex"], ["age"], ["eye_color"], "age|eye_color"),
+    ],
+    ids=[
+        "additional_no_operation",
+        "exclude_no_operation",
+        "additional_and_exclude_no_operation",
+    ],
+)
+def test_add_observation_nop_stratifications(default, additional, excluded, match):
+    ctx = ResultsContext()
+    ctx._default_stratifications = default
+    with pytest.warns(UserWarning, match=match):
```

Comment:
I suspect I just don't know how the pytest.warns context manager works, but what exactly happens if this fails?

Reply:
It complains that a UserWarning was expected but didn't get made.

Reply:
But don't you need to assert something about this or is that part of pytest.warns job?

Reply:
Nope. https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.warns